### PR TITLE
Fix parsing of EventData block for Windows Events.

### PR DESCRIPTION
**Pull request by zacbrown**
Fix parsing of EventData block for Windows Events. Previously, parsed EventData blocks from the Windows EventLog would contain two additional "non-real" fields:

Old event example (4688 - process start), **note the properties`<xmlattr>` and `Name`**:
```
{
    "EventData": {
        "<xmlattr>": "",
        "CommandLine": "osqueryi  --disable_events=false",
        "MandatoryLabel": "S-1-16-12288",
        "Name": "SubjectUserSid,SubjectUserName,SubjectDomainName,SubjectLogonId,NewProcessId,NewProcessName,TokenElevationType,ProcessId,CommandLine,TargetUserSid,TargetUserName,TargetDomainName,TargetLogonId,ParentProcessName,MandatoryLabel",
        "NewProcessId": "0x1d88",
        "NewProcessName": "C:\\ProgramData\\osquery\\osqueryi.exe",
        "ParentProcessName": "C:\\Windows\\System32\\cmd.exe",
        "ProcessId": "0x1e20",
        "SubjectDomainName": "DESKTOP-L8H2M4",
        "SubjectLogonId": "0x12345",
        "SubjectUserName": "Some Username",
        "SubjectUserSid": "S-1-5-21-1338873204-2012490888-2498879288-1000",
        "TargetDomainName": "-",
        "TargetLogonId": "0x0",
        "TargetUserName": "-",
        "TargetUserSid": "S-1-0-0",
        "TokenElevationType": "%%1937"
    }
}
```

Fixed output is as follows:
```
{
    "EventData": {
        "CommandLine": "C:/dev/oss/osquery/build/windows10/osquery/RelWithDebInfo/osqueryi.exe  --disable-events=false",
        "MandatoryLabel": "S-1-16-12288",
        "NewProcessId": "0x14ec",
        "NewProcessName": "C:\\dev\\oss\\osquery\\build\\windows10\\osquery\\RelWithDebInfo\\osqueryi.exe",
        "ParentProcessName": "C:\\Windows\\System32\\cmd.exe",
        "ProcessId": "0xa30",
        "SubjectDomainName": "DESKTOP-L8H2M4",
        "SubjectLogonId": "0x12345",
        "SubjectUserName": "Some User",
        "SubjectUserSid": "S-1-5-21-1338873204-2012490888-2498879288-1000",
        "TargetDomainName": "-",
        "TargetLogonId": "0x0",
        "TargetUserName": "-",
        "TargetUserSid": "S-1-0-0",
        "TokenElevationType": "%%1937"
    }
}
```

### CODE REVIEW

File: osquery/tables/events/windows/windows_events.cpp [hunk 1]
```diff
@@ -61,14 +61,25 @@ REGISTER(WindowsEventSubscriber, "event_subscriber", "windows_events");
...
       nodeName = node.first.empty() ? "DataElement" : node.first;
     }
-    res[nodeName] = res[nodeName] == ""
-                        ? node.second.data()
-                        : res[nodeName] + "," + node.second.data();
+
+    if (res[nodeName] == "") {
+      res[nodeName] = node.second.data();
+    }
+    else {
```
Comment (muffins) [inline osquery/tables/events/windows/windows_events.cpp]: nit: bring this up to the closing bracket. We should install `clang-format` on your dev box, and you can auto lint with `clang-format -i`

**Comment by muffins** [comment]
ok to test

**Comment by osqueryer** [comment]
:no_good: The commit 0c5b612e3d192132327934b2270ee1f6a6ebcf9a (Job results: [3405](https://jenkins.osquery.io/job/osqueryPullRequestAudit/3405/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit 1b9cde887c75970e0c847cfef98017f70e8b5473 (Job results: [6924](https://jenkins.osquery.io/job/osqueryPullRequestLinux/6924/)) failed one or more tests (**Linux**).

File: osquery/tables/events/windows/windows_events.cpp [hunk 2]
```diff
@@ -61,14 +61,24 @@ REGISTER(WindowsEventSubscriber, "event_subscriber", "windows_events");
...
   for (const auto& node : tree) {
-    auto nodeName = node.second.get("<xmlattr>.Name", "");
+    // Skip this since it's not actually part of the EventData. Also prevents
+    // us from adding every Name attribute into its own key invalidly. This is
+    // part of a quirk of boost::ptree and its parsing of XML.
+    if (node.first == "<xmlattr>") {
+      continue;
+    }
 
+    auto nodeName = node.second.get("<xmlattr>.Name", "");
```
Comment (obelisk) [inline osquery/tables/events/windows/windows_events.cpp]: Prefer snake case for variables

File: osquery/tables/events/windows/windows_events.cpp [hunk 3]
```diff
@@ -61,14 +61,24 @@ REGISTER(WindowsEventSubscriber, "event_subscriber", "windows_events");
...
     if (nodeName.empty()) {
       nodeName = node.first.empty() ? "DataElement" : node.first;
     }
-    res[nodeName] = res[nodeName] == ""
-                        ? node.second.data()
-                        : res[nodeName] + "," + node.second.data();
+
```
Comment (obelisk) [inline osquery/tables/events/windows/windows_events.cpp]: Why did you change this from a ternary?

File: osquery/tables/events/windows/windows_events.cpp (hunk 2 above)
Comment (muffins) [reply osquery/tables/events/windows/windows_events.cpp]: This is definitely the way it _should_ be done, however the entirety of the windows code base is currently incorrect :( I'd say for now stick with consistency and we can file a follow-up item to fix everything.

File: osquery/tables/events/windows/windows_events.cpp (hunk 3 above)
Comment (zacbrown) [reply osquery/tables/events/windows/windows_events.cpp]: In the initial change I made, I thought I was going to use something within the if statements. I left it as is after cleaning up the solution.

Happy to change it back if it's a stylistic thing.

Site note: I find ternary expressions hard to read, which is why I expanded it at all.

File: osquery/tables/events/windows/windows_events.cpp (hunk 2 above)
Comment (zacbrown) [reply osquery/tables/events/windows/windows_events.cpp]: Ya - I was imitating most of the other functions in the file but I did note that there's both snake case and pascal case floating around.
